datumaro/plugins/cifar_format.py

```diff
@@ -0,0 +1,179 @@
+
+        return items
+
+
+class CifarImporter(Importer):
+    @classmethod
+    def find_sources(cls, path):
+        return cls._find_sources_recursive(path, '', 'cifar',
+            file_filter=lambda p: osp.basename(p) != CifarPath.BATCHES_META and \
+                osp.basename(p) != CifarPath.IMAGES_DIR)
+
+
+class CifarConverter(Converter):
+    DEFAULT_IMAGE_EXT = '.png'
+
+    def apply(self):
+        label_categories = self._extractor.categories()[AnnotationType.label]
+
+        label_names = []
+        for label in label_categories:
+            label_names.append(label.name)
+        labels_dict = { 'label_names': label_names }
+        batches_meta_file = osp.join(self._save_dir, CifarPath.BATCHES_META)
+        with open(batches_meta_file, 'wb') as labels_file:
+            pickle.dump(labels_dict, labels_file)
+
+        for subset_name, subset in self._extractor.subsets().items():
+            filenames = []
+            labels = []
+            data = []
+            image_sizes = {}
+            for item in subset:
+                filenames.append(item.id + self._find_image_ext(item))
+
+                if item.has_image and self._save_images:
+                    self._save_image(item, osp.join(self._save_dir, CifarPath.IMAGES_DIR,
+                        self._make_image_filename(item)))
+
+                anns = [a.label for a in item.annotations
+                    if a.type == AnnotationType.label]
+                label = None
+                if anns:
+                    label = anns[0]
+                labels.append(label)
+
+                image = item.image.data
+                if not image.any():
+                    data.append(image)
+                else:
+                    data.append(image.reshape(image.shape[0] * image.shape[1] * \
+                        image.shape[2]).astype(np.uint8))
+                    if image.shape[0] != CifarPath.IMAGE_SIZE or \
+                            image.shape[1] != CifarPath.IMAGE_SIZE:
+                        image_sizes[len(data) - 1] = (image.shape[0], image.shape[1])
+
+            annotation_dict = {}
+            annotation_dict['filenames'] = filenames
+            annotation_dict['labels'] = labels
+            annotation_dict['data'] = np.array(data)
+            if len(image_sizes):
+                size = (CifarPath.IMAGE_SIZE, CifarPath.IMAGE_SIZE)
+                # 'image_sizes' isn't included in the standart format,
+                # needed for different image sizes
+                annotation_dict['image_sizes'] = [image_sizes.get(p, size)
+                    for p in range(len(data))]
+
+            filename = '%s_batch' % subset_name
+            batch_label = None
+            if subset_name.startswith('train_'):
+                num = subset_name.split('_')[1]
+                filename = CifarPath.TRAIN_ANNOTATION_FILE + num
+                batch_label = 'training batch %s of 5' % num
```

Comment:
Is there a fixed total number of batches?

Reply:
Yes, dataset is divided into five training batches and one test batch.

Reply:
It is not guaranteed, that symbols after `_` will be a number. I suggest adding a check with `datumato.utils.cast` at least.